common/src/main/java/com/microsoft/identity/common/internal/commands/DeviceCodeFlowCommand.java

```diff
@@ -47,8 +59,30 @@ public DeviceCodeFlowCommand(@NonNull DeviceCodeFlowCommandParameters parameters
 
     @Override
     public AcquireTokenResult execute() throws Exception {
-        // TODO: Implement execute body in next PR, when DCF methods have been declared/defined in LocalMsalController
-        return null;
+
+        // Get the controller used to execute the command
+        final BaseController controller = getDefaultController();
```

Comment:
This is wrong as stated the CR on an earlier PR. This would use the `BrokerMsalController` if a broker is present on the device. 

This should instead be `getControllers()` and loop over them to execute for each controller. For an example, see `SilentTokenCommand.java`

Reply:
This is not supported in broker. so either we find a way to inject LocalMsalController as the default controller elsewhere, or we simply just hard code it here and add some comment.

Reply:
Then we should use `getControllers().get(0)` - The command object has at least one controller to begin with. In this case, that is the `LocalMsalController` as that is being supplied to the command during creation via the method in PCA

Reply:
@shahzaibj `getDefaultController()` here actually points to a method in `BaseController` that simply calls `controllers.get(0)` (`controllers` being the object that would be returned by `getControllers()`), so it's the same code either way I think. And yes, since the command object was created with a `LocalMsalController` in `PublicClientApplication`, `LocalMsalController` should be the "default" controller.

Reply:
@t-fadura Okay, I see. We have multiple methods called `getDefaultController` and each of them do different things. Thanks for pointing out.